Added a wiki to outline the triage process. (#51)
oppia/oppia-web-developer-docs: Triaging-process.md

## General
This wiki outlines the steps to triage new and existing issues. This ensures that all issues are being effectively tracked and prioritized.

### Steps to triage incoming issues coming into Oppia as a whole (first-stage triage):
The Dev Workflow team lead (currently Chris) should do the following every week for any [filed issues that aren’t assigned to a project](https://github.com/oppia/oppia/issues?q=is%3Aissue+is%3Aopen+no%3Aproject):

- Check that the issue is written clearly enough. Request clarification if needed.
- Assign the issue to the appropriate task force, if applicable. Otherwise, assign it to the relevant team (LaCE, CD, Dev Workflow) based on which type of user it affects.
- If you’re not sure how to categorise an issue, feel free to ping Sean for help!

### Steps to triage incoming issues for each team (second-stage triage):
Team leads should do an audit every week to categorise new issues in their GitHub Projects board, following these steps:

1. Verify that the issue lies within the scope of the team (see definitions below).
    - If not, assign the issue to the appropriate project/team.
2. Determine whether the issue is a bug or a enhancement

>     - If the issue is a feature request (enhancement), contact the relevant PM (product manager) for the team and ask them to take a look and decide whether it needs to be worked upon. (See the list of PMs [below](##pm-contacts-for-each-team).)

Reviewer: I don't think the `[below](##pm-contacts-for-each-team)` link is working please fix it.

Author: Fixed

3. (For simple UI issues) Make sure the bug is reproducible
    - The issue should be consistently reproducible on either develop or the test/backup server.
    - If that isn’t the case, ask the issue author for clarity.
4. Once the issue has been confirmed, the team lead should add the  appropriate ‘Impact’ label (asking the PM or tech lead for help if they’re not sure), as well as the appropriate ‘Work’ label to the issue.

>     - ‘Impact’ defines how much the issue affects the functionality of the user-facing features. There are 3 associated labels: **Impact: High**, **Impact: Medium** and **Impact: Low**. For eg., the issue is considered to have a high impact if it breaks a critical feature or blocks a release. General bugs (non-blocking) and improvements can be considered to be of medium impact while issues that occur in a few edge cases or affect a small number of users are issues with a low impact.

Reviewer 2: For eg. --> For example

Author: Done

    ||||||
    |--- |--- |--- |--- |--- |
    ||Severity|Severe (blocks or represents a primary user journey – e.g. end user playing a lesson, developer installing Oppia)|Moderate severity (blocks or represents a secondary user journey – e.g. end user changing a profile picture, developer easily seeing the output of frontend tests at the bottom of the CI log)|Minor (neither of the others, relates to “polish”)|
    |Users Affected|||||
    |(roughly) 60+%||High|Medium|Low|
    |(roughly) 5-60%||High|Low|Low|

>     |(roughly) < 5%||Medium|Low|Low|

Reviewer: Please fix the table.

Author: Done

>     <p style="text-align:center; font-size:12px"> Table: How “impact” is determined </p>

Reviewer: Do we need this to be in the center? If so, please fix it.

Author: Done

    - ‘Work’ defines how hard it would be to solve/resolve a particular issue. This is generally correlated with the amount of ambiguity in the task. We use the following rough system for assigning this label:
        - Solution is clear and broken into good-first-issue-sized chunks – assign **Work: Low**
        - Otherwise, if the means to find the solution is clear – assign **Work: Medium**
        - Everything else – assign **Work: High**

5. Once the appropriate ‘Impact’ and ‘Work Required’ labels has been added, determine its severity and assign the fitting priority label based on its impact (and possibly the work required label)
    - If the issue is a high-impact bug, put it in the high-priority bucket and prioritise it within the team.

>     - For the priority bucket, team leads should include “important” issues – typically, a selection of high-impact enhancements and low-work medium-impact bugs – while keeping the size of that bucket about 2-3 times the projected stable number of team members. (Note: This exercise might also help with determining how large the team itself needs to be.)

Reviewer: ```suggestion
    - For the priority bucket, team leads should include “important” issues – typically, a selection of high-impact enhancements and low-work medium-impact bugs – while keeping the size of that bucket about 2-3 times the projected stable number of team members.
    > **Note**
    > This exercise might also help with determining how large the team itself needs to be.
```

    - Other issues go in the ‘backlog’ bucket.

>     - Any issue with an external dependency that the team cannot resolve by itself (eg. missing mocks) goes in the ‘blocked’ bucket, and the reason should be specified on the ‘Issue specific status’ field (using free-text). Team leads should make sure an issue spends as little time here as possible.

Reviewer: ```suggestion
    - Any issue with an external dependency that the team cannot resolve by itself (example: missing mocks) goes in the ‘blocked’ bucket, and the reason should be specified in the ‘Issue specific status’ field (using free-text). Team leads should ensure an issue spends as little time here as possible.
```

6. Determine if the issue is potentially a good first issue
    - If the issue seems like something a new contributor to Oppia can take up, make sure to add enough information in the issue description for a new contributor to start work on that issue then ask Sandeep or Vojta to add the Good First Issue label. (Note: typically these are “Work: Low” issues.)

>         - For such issues, also assign purple “technology: TypeScript”, “technology: Apache Beam”, “technology: Python”, etc. labels as appropriate. This helps new contributors identify issues that they would find approachable.

Reviewer: What type of issues? (Good first issues?)

Author: Yes these refer to good first issues. Should I explicitly mention that, as they are already nested under a related point?

Reviewer: Just to rule out any possibility of confusion. change `For such issues` to `For such good first issues`

7. After triaging all issues, make sure the issues in the 'Priority' and 'High Priority' buckets are sorted based on severity/priority (high to low).

## PM contacts for each team
- **LaCE Quality**: @dchen97 (learners), @seanlip (creators)
- **Contributor dashboard**: @dchen97
- **Dev workflow**: @U8NWXD

> - **Android**: @seanlip

Reviewer: Sorry for not being clear. Can you mention it in this format: "<Name> (GitHub_ID)"

Author: Done

- Check that the issue is written clearly enough. Request clarification if needed.
- Assign the issue to the appropriate task force, if applicable. Otherwise, assign it to the relevant team (LaCE, CD, Dev Workflow) based on which type of user it affects.
- If you’re not sure how to categorise an issue, feel free to ping Sean (@seanlip) for help!

### Second-stage triage - Steps to triage incoming issues for each team:
Team leads should do an audit every week to categorise new issues in their GitHub Projects board, following these steps:

1. Verify that the issue lies within the scope of the team (see definitions below).
    - If not, assign the issue to the appropriate project/team.
2. Determine whether the issue is a bug or a enhancement
    - If the issue is a feature request (enhancement), contact the relevant PM (product manager) for the team and ask them to take a look and decide whether it needs to be worked upon. (See the list of PMs [below](#pm-contacts-for-each-team).)

> 3. (For simple UI issues) Make sure the bug is reproducible

Reviewer: In case it is not a UI issue then what should the devs do?

Reviewer: Try and mention the step the devs should take and list the exception or points to be noted as subpoints.

Author: Done, PTAL.

        || Severity  | Severe (blocks or represents a primary user journey – e.g. end user playing a lesson, developer installing Oppia) | Moderate severity (blocks or represents a secondary user journey – e.g. end user changing a profile picture, developer easily seeing the output of frontend tests at the bottom of the CI log) | Moderate severity (blocks or represents a secondary user journey – e.g. end user changing a profile picture, developer easily seeing the output of frontend tests at the bottom of the CI log) |
        | -------------------  | ------------------- | --------------------- | ------------------| ---------- |
        | **# users affected**|        |        |     |
        | (roughly) 60+%      || High   | Medium | Low |
        | (roughly) 5-60%     || High   | Low    | Low |

>         | (roughly) < 5%      || Medium | Low    | Low |

Reviewer: This seems to be a lot of information to take in, especially the headers. Can we simplify the format?

Reviewer: Also, the `Moderate severity` seems to be repeated twice. is this intentional?

Author: Fixed.

- If you’re not sure how to categorise an issue, feel free to ping Sean (@seanlip) for help!

### Second-stage triage - Steps to triage incoming issues for each team:
Team leads should do an audit every week to categorise new issues in their GitHub Projects board, following these steps:

1. Verify that the issue lies within the scope of the team (see definitions below).
    - If not, assign the issue to the appropriate project/team.
2. Determine whether the issue is a bug or a enhancement
    - If the issue is a feature request (enhancement), contact the relevant PM (product manager) for the team and ask them to take a look and decide whether it needs to be worked upon. (See the list of PMs [below](#pm-contacts-for-each-team).)
    - If the issue is a bug, add the **Bug** label.
3. (For simple UI issues) Make sure the bug is reproducible
    - The issue should be consistently reproducible on either develop or the test/backup server.
    - If that isn’t the case, ask the issue author for clarity.
    - If the issue is not a simple UI bug that can be easily verified, the task of verifying the reproducibility lies on the developer working on the issue.

> 4. Once the issue has been confirmed, the team lead should add the  appropriate ‘Impact’ label (asking the PM or tech lead for help if they’re not sure), as well as the appropriate ‘Work’ label to the issue.

Reviewer: What are the criteria for an issue getting confirmed? Do you mean "Once the issue has been created"?

Author: I have made some changes. PTAL.

## General
This wiki outlines the steps to triage new and existing issues. This ensures that all issues are being effectively tracked and prioritized.

> ### Steps to triage incoming issues coming into Oppia as a whole (first-stage triage):

Reviewer: Probably mention "first-stage triage" at the beginning. So the reader knows that there are 2 stages.

Author: Done

The Dev Workflow team lead (currently Chris) should do the following every week for any [filed issues that aren’t assigned to a project](https://github.com/oppia/oppia/issues?q=is%3Aissue+is%3Aopen+no%3Aproject):

- Check that the issue is written clearly enough. Request clarification if needed.
- Assign the issue to the appropriate task force, if applicable. Otherwise, assign it to the relevant team (LaCE, CD, Dev Workflow) based on which type of user it affects.
- If you’re not sure how to categorise an issue, feel free to ping Sean (@seanlip) for help!

> ### Steps to triage incoming issues for each team (second-stage triage):

Reviewer: *Ditto

Author: Done

    - If that isn’t the case, ask the issue author for clarity.
4. Once the issue has been confirmed, the team lead should add the  appropriate ‘Impact’ label (asking the PM or tech lead for help if they’re not sure), as well as the appropriate ‘Work’ label to the issue.
    - ‘Impact’ defines how much the issue affects the functionality of the user-facing features. There are 3 associated labels: **Impact: High**, **Impact: Medium** and **Impact: Low**. For example, the issue is considered to have a high impact if it breaks a critical feature or blocks a release. General bugs (non-blocking) and improvements can be considered to be of medium impact while issues that occur in a few edge cases or affect a small number of users are issues with a low impact.

        - Severity:
            - **Severe**: Blocks or represents a primary user journey – e.g. end user playing a lesson, developer installing Oppia.
            - **Moderate**: blocks or represents a secondary user journey – e.g. end user changing a profile picture, developer easily seeing the output of frontend tests at the bottom of the CI log.
            - **Minor**: Neither of the others, relates to “polish”.

>         <br />

Reviewer: Do we need this?

Author: I'm not really sure, but I think this provides the reader about the 'scope' of multiple levels of severity.

Reviewer: Sorry for not being clear. I meant the ` <br />` tag.

Author: I have replaced the table with an image. The `<br />` tag is no longer needed.

        
        |     | Severity       | Severe | Moderate | Minor |
        | --- | -------------- | ------ | -------- | ----- |
        | **# users affected** |        |          |       |
        | (roughly) 60+%      || High   | Medium   | Low   |
        | (roughly) 5-60%     || High   | Low      | Low   |
        | (roughly) < 5%      || Medium | Low      | Low   |

        Table: How “impact” is determined.

>         <br />

Reviewer: Do we need this?

Author: I think we would need some way to tell the readers the use of this table.

Reviewer: Sorry for not being clear. I meant the ` <br />` tag.

Author: I have replaced the table with an image. The `<br />` tag is no longer needed.

    - ‘Work’ defines how hard it would be to solve/resolve a particular issue. This is generally correlated with the amount of ambiguity in the task. We use the following rough system for assigning this label:
        - Solution is clear and broken into good-first-issue-sized chunks – assign **Work: Low**
        - Otherwise, if the means to find the solution is clear – assign **Work: Medium**

>         - Everything else – assign **Work: High**

Reviewer: Mention the `Work` label at the beginning of the point and then explain when to assign it.

Author: Done

5. Once the appropriate ‘Impact’ and ‘Work Required’ labels has been added, determine its severity and assign the fitting priority label based on its impact (and possibly the work required label)
    - If the issue is a high-impact bug, put it in the high-priority bucket and prioritise it within the team.
    - For the priority bucket, team leads should include “important” issues – typically, a selection of high-impact enhancements and low-work medium-impact bugs – while keeping the size of that bucket about 2-3 times the projected stable number of team members.

>     > **Note**
>     > This exercise might also help with determining how large the team itself needs to be.
>     - Other issues go in the ‘backlog’ bucket.
>     - Any issue with an external dependency that the team cannot resolve by itself (example: missing mocks) goes in the ‘blocked’ bucket, and the reason should be specified in the ‘Issue specific status’ field (using free-text). Team leads should ensure an issue spends as little time here as possible.

Reviewer: ```suggestion
      > **Note**
      > This exercise might also help determine how large the team needs to be.

    - Other issues go in the ‘backlog’ bucket.
    - Any issue with an external dependency that the team cannot resolve by itself (example: missing mocks) goes in the ‘blocked’ bucket, and the reason should be specified in the ‘Issue specific status’ field (using free-text). Team leads should ensure an issue spends as little time here as possible.
```

## General
This wiki outlines the steps to triage new and existing issues. This ensures that all issues are being effectively tracked and prioritized.

### Steps to triage incoming issues coming into Oppia as a whole (first-stage triage):
The Dev Workflow team lead (currently Chris) should do the following every week for any [filed issues that aren’t assigned to a project](https://github.com/oppia/oppia/issues?q=is%3Aissue+is%3Aopen+no%3Aproject):

- Check that the issue is written clearly enough. Request clarification if needed.
- Assign the issue to the appropriate task force, if applicable. Otherwise, assign it to the relevant team (LaCE, CD, Dev Workflow) based on which type of user it affects.
- If you’re not sure how to categorise an issue, feel free to ping Sean for help!

### Steps to triage incoming issues for each team (second-stage triage):
Team leads should do an audit every week to categorise new issues in their GitHub Projects board, following these steps:

> - Verify that the issue lies within the scope of the team (see definitions below).

Reviewer: Please number the steps

Author: Done

    - If not, assign the issue to the appropriate project/team.
- Determine whether the issue is a bug or a enhancement

>     - If the issue is a feature request (enhancement), contact the relevant PM (product manager) for the team and ask them to take a look and decide whether it needs to be worked upon. (See the list of PMs [below].)

Reviewer 2: Can you make "below" a link?

Author: Done

- (For simple UI issues) Make sure the bug is reproducible
    - The issue should be consistently reproducible on either develop or the test/backup server.

>     - If that isn’t the case ask the issue author for clarity

Reviewer 2: If that isn't the case, ask the issue author for clarity.

(missing comma and period)

Author: Done

    <table>
        <tr>
            <td></td>
            <td><b>Severity</b></td>
            <td>Severe (blocks or represents a primary user journey – e.g. end user playing a lesson, developer installing Oppia)</td>
            <td>Moderate severity (blocks or represents a secondary user journey – e.g. end user changing a profile picture, developer easily seeing the output of frontend tests at the bottom of the CI log)</td>
            <td>Minor (neither of the others, relates to “polish”)</td>
        </tr>
        <tr>
            <td><b>Users Affected</b></td>
            <td></td>
            <td></td>
            <td></td>
            <td></td>
        </tr>
        <tr>
            <td>(roughly) 60+%</td>
            <td></td>
            <td>High</td>
            <td>Medium</td>
            <td>Low</td>
        </tr>
        <tr>
            <td>(roughly) 5-60%</td>
            <td></td>
            <td>High</td>
            <td>Low</td>
            <td>Low</td>
        </tr>
        <tr>
            <td>(roughly) &lt; 5%</td>
            <td></td>
            <td>Medium</td>
            <td>Low</td>
            <td>Low</td>
        </tr>
    </table>

>     <p style="text-align:center; font-size:12px"> Table: How “impact” is determined </p>

Reviewer: Can you try using the [markdown table syntax](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-tables)?

Author: Done

    - ‘Work’ defines how hard it would be to solve/resolve a particular issue. This is generally correlated with the amount of ambiguity in the task. We use the following rough system for assigning this label:

>         - Solution is clear and broken into good-first-issue-sized chunks – assign Work: Low

Reviewer 2: Here and elsewhere, make label names (e.g. "Work: Low") bold

Author: Done

        - Otherwise, if the means to find the solution is clear – assign Work: Medium
        - Everything else – assign Work: High

> - Once the issue has been confirmed and the appropriate ‘Impact’ and ‘Work Required’ has been added, determine its severity and assign the fitting priority label based on its impact (and possibly the work required label)

Reviewer 2: ...'Work Required' labels have been added, ...

Author: Done

    - Other issues go in the ‘backlog’ bucket.
    - Any issue with an external dependency that the team cannot resolve by itself (eg. missing mocks) goes in the ‘blocked’ bucket, and the reason should be specified on the ‘Issue specific status’ field (using free-text). Team leads should make sure an issue spends as little time here as possible.

- Determine if the issue is potentially a good first issue
    - If the issue seems like something a new contributor to Oppia can take up, make sure to add enough information in the issue description for a new contributor to start work on that issue then ask Sandeep or Vojta to add the Good First Issue label. (Note: typically these are “Work: Low” issues.)
        - For such issues, also assign purple “technology: TypeScript”, “technology: Apache Beam”, “technology: Python”, etc. labels as appropriate. This helps new contributors identify issues that they would find approachable.

> After triaging all issues, make sure the issues in the Priority and High Priority Buckets are sorted based on severity/priority (high to low).

Reviewer: Is this a step or a note?

Author: It is a step, made the necessary changes.

## PM contacts for each team
- **LaCE Quality**: Diana (learners), Sean (creators)
- **Contributor dashboard**: Diana
- **Dev workflow**: Chris

> - **Android**: Sean

Reviewer: Try mentioning the GitHub id, so it will be easier for users to identify the GitHub ID.

Author: Done
